python_modules/libraries/dagster-dg/dagster_dg/cli/plus/deploy_session.py

```diff
@@ -0,0 +1,210 @@
+        f"{registry}:{tag}",
+        "-f",
+        dockerfile_path,
+        "--platform",
+        "linux/amd64",
+    ]
+
+    if use_editable_dagster:
+        build_cmd += [
+            "--build-context",
+            f"oss={os.environ['DAGSTER_GIT_REPO_DIR']}",
+            "--build-context",
+            f"internal={os.environ['DAGSTER_INTERNAL_GIT_REPO_DIR']}",
+        ]
+
+    subprocess.run(build_cmd, check=True)
+
+    push_cmd = [
+        "docker",
+        "push",
+        f"{registry}:{tag}",
+    ]
+
+    subprocess.run(push_cmd, check=True)
+
+    dg_context.external_dagster_cloud_cli_command(
+        [
+            "ci",
+            "set-build-output",
+            "--statedir",
+            str(statedir),
+            "--location-name",
+            dg_context.code_location_name,
+            "--image-tag",
+            tag,
+        ]
+    )
+
+
+def init_deploy_session(
+    organization: str,
+    deployment: str,
+    dg_context: DgContext,
+    statedir: str,
+    input_deployment_type: Optional[DgPlusDeploymentType],
+    skip_confirmation_prompt: bool,
+):
+    deployment_type = (
+        input_deployment_type
+        if input_deployment_type
+        else _guess_and_prompt_deployment_type(
+            dg_context.root_path, deployment, skip_confirmation_prompt
+        )
+    )
+
+    with create_temp_dagster_cloud_yaml_file(dg_context) as dagster_cloud_yaml_file:
+        dg_context.external_dagster_cloud_cli_command(
+            [
+                "ci",
+                "init",
+                "--statedir",
+                str(statedir),
+                "--dagster-cloud-yaml-path",
+                dagster_cloud_yaml_file,
+                "--project-dir",
+                str(dg_context.root_path),
+                "--deployment",
+                deployment,
+                "--organization",
+                organization,
+                "--clean-statedir",
+            ]
+            + (
+                ["--require-branch-deployment"]
+                if deployment_type == DgPlusDeploymentType.BRANCH_DEPLOYMENT
+                else []
+            ),
+        )
+
+
+def build_artifact(
+    dg_context: DgContext,
+    agent_type: DgPlusAgentType,
+    statedir: str,
+    use_editable_dagster: bool,
+    python_version: Optional[str],
+):
+    if not python_version:
+        python_version = f"3.{sys.version_info.minor}"
+
+    dockerfile_path = dg_context.root_path / "Dockerfile"
+    if not os.path.exists(dockerfile_path):
+        click.echo(f"No Dockerfile found - scaffolding a default one at {dockerfile_path}.")
+        _create_temp_deploy_dockerfile(dockerfile_path, python_version, use_editable_dagster)
+    else:
+        click.echo(f"Building using Dockerfile at {dockerfile_path}.")
+
+    if agent_type == DgPlusAgentType.HYBRID:
+        _build_hybrid_image(dg_context, dockerfile_path, use_editable_dagster, statedir)
+
+    else:
+        dg_context.external_dagster_cloud_cli_command(
```

Comment:
This makes PEX out of scope for now

Reply:
i'll file a todo to add it once the pyproject.toml support is in